dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -792,13 +792,20 @@ public List<Tool> allPublishedContainers(
         @ApiParam(value = "Sort column") @DefaultValue("stars") @QueryParam("sortCol") String sortCol,
         @ApiParam(value = "Sort order", allowableValues = "asc,desc") @DefaultValue("desc") @QueryParam("sortOrder") String sortOrder,
         @Context HttpServletResponse response) {
-        int maxLimit = Math.min(Integer.parseInt(PAGINATION_LIMIT), limit);
-        List<Tool> tools = toolDAO.findAllPublished(offset, maxLimit, filter, sortCol, sortOrder);
-        filterContainersForHiddenTags(tools);
-        stripContent(tools);
-        response.addHeader("X-total-count", String.valueOf(toolDAO.countAllPublished(Optional.of(filter))));
-        response.addHeader("Access-Control-Expose-Headers", "X-total-count");
-        return tools;
+        try {
+            int maxLimit = Math.min(Integer.parseInt(PAGINATION_LIMIT), limit);
+            List<Tool> tools = toolDAO.findAllPublished(offset, maxLimit, filter, sortCol, sortOrder);
+            filterContainersForHiddenTags(tools);
+            stripContent(tools);
+            response.addHeader("X-total-count", String.valueOf(toolDAO.countAllPublished(Optional.of(filter))));
+            response.addHeader("Access-Control-Expose-Headers", "X-total-count");
+            return tools;
+        } catch (IllegalArgumentException e) {
```

Comment:
Following up on Denis' comment, I'm guessing you get this Exception if sortOrder is invalid as per the bug. What if sort order is valid, but something is wrong with the filter? Do you get the same exception? I don't know.

Suggest have tests for both cases (bad sort order, bad filter).

Reply:
After Kathy's review comment, I confirmed that sortCol is the only one that can throw the error